File: src/Other/PageInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenSilver.Samples.Showcase
{
    public class PageInfo
    {
        public PageInfo(string name, string path, bool isVisibleInMenu) => (Name, Path, IsVisibleInMenu) = (name, path, isVisibleInMenu);

        static List<PageInfo> _pageInfos;
        static PageInfo _landingPageInfo;
        static PageInfo _searchPageInfo;
        public static List<PageInfo> Pages
        {
            get
            {
                if (_pageInfos == null)
                {
                    _pageInfos = new List<PageInfo>();
                    _pageInfos.Add(new PageInfo("Panels & Controls", "/XAML_Controls", true));
                    _pageInfos.Add(new PageInfo("Xaml Features", "/XAML_Features", true));
                    _pageInfos.Add(new PageInfo(".NET Framework", "/Net_Framework", true));
                    _pageInfos.Add(new PageInfo("Client / Server", "/Client_Server", true));
                    _pageInfos.Add(new PageInfo("Interop", "/Interop_Samples", true));
                    _pageInfos.Add(new PageInfo("JS Libs", "/JS_Libs", true));
                    _pageInfos.Add(new PageInfo("Charts", "/Charts", true));
                    _pageInfos.Add(new PageInfo("Performance", "/Performance", true));
                    _pageInfos.Add(new PageInfo("Native APIs", "/Maui_Hybrid", true));
                    _pageInfos.Add(new PageInfo("Third-Party", "/Third_Party", true));
                    _pageInfos.Add(LandingPageInfo);
                    _pageInfos.Add(SearchPageInfo);
                }

                return _pageInfos;
            }
        }

        public static PageInfo LandingPageInfo
        {
            get
            {
                if (_landingPageInfo == null)
                {
                    _landingPageInfo = new PageInfo("Home", "/Welcome", false);
                }
                return _landingPageInfo;
            }
        }

        public static PageInfo SearchPageInfo
        {
            get
            {
                if (_searchPageInfo == null)
                {
                    _searchPageInfo = new PageInfo("Search", "/Search", false);
                }
                return _searchPageInfo;
            }
        }


        public string Name { get; set; }
        public string Path { get; set; }
        public bool IsVisibleInMenu { get; set; }
    }
}

File: src/Other/PageInfo.vb
Namespace OpenSilver.Samples.Showcase

    Public Class PageInfo

        Public Sub New(name As String, path As String, isVisibleInMenu As Boolean)
            Me.Name = name
            Me.Path = path
            Me.IsVisibleInMenu = isVisibleInMenu
        End Sub

        Private Shared _pageInfos As List(Of PageInfo)
        Private Shared _landingPageInfo As PageInfo
        Private Shared _searchPageInfo As PageInfo

        Public Shared ReadOnly Property Pages As List(Of PageInfo)
            Get
                If _pageInfos Is Nothing Then
                    _pageInfos = New List(Of PageInfo) From {
                        New PageInfo("Panels & Controls", "/XAML_Controls", True),
                        New PageInfo("Xaml Features", "/XAML_Features", True),
                        New PageInfo(".NET Framework", "/Net_Framework", True),
                        New PageInfo("Client / Server", "/Client_Server", True),
                        New PageInfo("Interop", "/Interop_Samples", True),
                        New PageInfo("JS Libs", "/JS_Libs", True),
                        New PageInfo("Charts", "/Charts", True),
                        New PageInfo("Performance", "/Performance", True),
                        New PageInfo("Native APIs", "/Maui_Hybrid", True),
                        New PageInfo("Third-Party", "/Third_Party", True),
                        LandingPageInfo,
                        SearchPageInfo
                    }
                End If
                Return _pageInfos
            End Get
        End Property

        Public Shared ReadOnly Property LandingPageInfo As PageInfo
            Get
                If _landingPageInfo Is Nothing Then
                    _landingPageInfo = New PageInfo("Home", "/Welcome", False)
                End If
                Return _landingPageInfo
            End Get
        End Property

        Public Shared ReadOnly Property SearchPageInfo As PageInfo
            Get
                If _searchPageInfo Is Nothing Then
                    _searchPageInfo = New PageInfo("Search", "/Search", False)
                End If
                Return _searchPageInfo
            End Get
        End Property

        Public Property Name As String
        Public Property Path As String
        Public Property IsVisibleInMenu As Boolean

    End Class

End Namespace
